src/Swarm/TUI/Controller.hs

```diff
@@ -747,6 +747,15 @@ handleREPLEvent = \case
       CmdPrompt {} -> continueWithoutRedraw
       SearchPrompt _ _ ->
         uiState %= resetWithREPLForm (mkReplForm $ mkCmdPrompt "")
+  ControlKey 'd' -> do
+    formSt <- use $ uiState . uiReplForm . to formState
+    case formSt of
+      CmdPrompt t _ -> if t == T.empty
+        then toggleModal QuitModal
+        else continueWithoutRedraw
+      SearchPrompt t _ -> if t == T.empty
+        then toggleModal QuitModal
+        else continueWithoutRedraw
```

Comment:
There is a lens `promptTextL` that you can use to avoid duplicating code. :wink:

Reply:
Ah great, I see it now, thanks! :)

Reply:
@0xcefaedfe since you said you were new to Haskell I'm not sure how familiar you are with lenses.  Lenses are amazing but can have a steep learning curve. :smile:  In this case, you should be able to just be able to write e.g. `formSt ^. promptTextL` to refer to the text of the prompt.  Or you could just add it to the end of line 751 to get the prompt text directly: `txt <- use $ uiState . uiReplForm . to formState . promptTextL`.

Reply:
@byorgey I think he already did :wink: